lib/moon/helpers/surface_render.ex

```diff
@@ -0,0 +1,57 @@
+defmodule Moon.Helpers.SurfaceRender do
+  @moduledoc """
+  Functions for rendering Surface components in HEEX/SLIM/... (not Surface) templates
+  """
+
+  import Phoenix.LiveView.Helpers, only: [component: 2, live_component: 1]
+
+  defp get_default_props(module) do
+    Enum.reduce(module.__props__(), %{__context__: %{}}, fn
+      %{name: name, opts: opts}, acc ->
+        Map.put(acc, name, Keyword.get(opts, :default))
+    end)
+  end
+
+  defp transform_slots(
+         props = %{inner_block: [%{__slot__: :inner_block, inner_block: inner_block}]}
+       ) do
+    props
+    |> Map.delete(:inner_block)
+    |> Map.put(:default, [%{__slot__: :default, inner_block: inner_block}])
+  end
+
+  defp transform_slots(props), do: props
+
+  defp render_stateless_component(props = %{module: module}) do
+    component(
+      &module.render/1,
+      module
+      |> get_default_props()
+      |> Map.merge(props |> Map.delete(:module) |> transform_slots())
+    )
+  end
+
+  defp render_stateful_component(props = %{module: module}) do
+    module |> get_default_props() |> Map.merge(props |> transform_slots()) |> live_component()
+  end
+
+  defp get_render_function(module) do
+    %{kind: :component} = module.__live__()
+    &render_stateful_component/1
+  rescue
+    [UndefinedFunctionError, MatchError] ->
+      &render_stateless_component/1
+  end
+
+  @doc """
+  A function for rendering Surface components inside non-surface templates
+  usage: `<.surface module="Button">hit me!</.surface>`
```

Comment:
How this differs from 

`Moon.RenderHelpers`

And why it is better?

Reply:
1. we need function with arity=1 to use it as a component
2. Moon.RenderHelpers.get_default_props collects :values property, not :default, not sure why. Looks like kind of breaking changes for me
3. inner_content => slot name: default 
4. it does recognise is the component state{ful/less} by itself, s
and also wanted to make a short name to look nice in templates

Reply:
and yeah, it's just refactored copy of `Moon.RenderHelpers`

Reply:
I think we should have only 1 version, current setup is weird. a xor b.

Additionally, I am not certain about `.surface module=` part. 

It should expose 2 render functions for live and stateless. ".surface" is meaningless when I look at it. "render_surface_stateless" and "render_surface_stateful" would be better, but maybe we just have them in namespace?